packages/dd-trace/test/util.spec.js

```diff
@@ -86,3 +86,177 @@ describe('generatePointerHash', () => {
+    })
+
+    it('returns empty buffer for unsupported type', () => {
+      const result = encodeValue({ A: 'abc' })
+      expect(Buffer.isBuffer(result)).to.be.true
+      expect(result.length).to.equal(0)
+    })
+
+    it('returns empty buffer for malformed input', () => {
+      const result = encodeValue({})
+      expect(Buffer.isBuffer(result)).to.be.true
+      expect(result.length).to.equal(0)
+    })
+
+    it('handles empty string values', () => {
+      const result = encodeValue({ S: '' })
+      expect(Buffer.isBuffer(result)).to.be.true
+      expect(result.length).to.equal(0)
+    })
+
+    it('handles empty buffer', () => {
+      const result = encodeValue({ B: Buffer.from([]) })
+      expect(Buffer.isBuffer(result)).to.be.true
+      expect(result.length).to.equal(0)
+    })
+  })
+})
+
+describe('extractPrimaryKeys', () => {
+  describe('single key table', () => {
+    it('handles string key with Set input', () => {
+      const keySet = new Set(['userId'])
+      const item = { userId: { S: 'user123' } }
+      const result = extractPrimaryKeys(keySet, item)
+      expect(result).to.deep.equal(['userId', Buffer.from('user123'), '', ''])
+    })
+
+    it('handles number key with Set input', () => {
+      const keySet = new Set(['timestamp'])
+      const item = { timestamp: { N: '1234567' } }
+      const result = extractPrimaryKeys(keySet, item)
+      expect(result).to.deep.equal(['timestamp', Buffer.from('1234567'), '', ''])
+    })
+
+    it('handles binary key with Set input', () => {
+      const keySet = new Set(['binaryId'])
+      const binaryData = Buffer.from([1, 2, 3])
+      const item = { binaryId: { B: binaryData } }
+      const result = extractPrimaryKeys(keySet, item)
+      expect(result).to.deep.equal(['binaryId', binaryData, '', ''])
+    })
+
+    it('handles string key with Object input', () => {
+      const keyObj = { userId: { S: 'user123' } }
+      const result = extractPrimaryKeys(keyObj, keyObj)
+      expect(result).to.deep.equal(['userId', Buffer.from('user123'), '', ''])
+    })
+  })
+
+  describe('double key table', () => {
+    it('handles and sorts string-string composite key', () => {
+      const keySet = new Set(['userId', 'email'])
+      const item = {
+        userId: { S: 'user123' },
+        email: { S: 'test@example.com' }
+      }
+      const result = extractPrimaryKeys(keySet, item)
+      expect(result).to.deep.equal(['email', Buffer.from('test@example.com'), 'userId', Buffer.from('user123')])
+    })
+
+    it('handles and sorts string-number composite key', () => {
+      const keySet = new Set(['timestamp', 'userId'])
+      const item = {
+        timestamp: { N: '1234567' },
+        userId: { S: 'user123' }
+      }
+      const result = extractPrimaryKeys(keySet, item)
+      expect(result).to.deep.equal(['timestamp', Buffer.from('1234567'), 'userId', Buffer.from('user123')])
+    })
+
+    it('handles and sorts composite key with Object input', () => {
```

Comment:
do these sort tests behave differently if the sorting step is omitted from our code? no big deal if not, i guess, just wondering what the underlying key order behavior actually is.

Reply:
Yes, the keys are not guaranteed to be in order, so we have to sort them first